Refresh NuGet get started articles for VS 2022 #1 (#2850)
NuGet/docs.microsoft.com-nuget: docs/quickstart/install-and-use-a-package-in-visual-studio.md

---
title: "Quickstart: Install and use a NuGet package in Visual Studio (Windows only)"
description: In this quickstart, you learn how to install and use a NuGet package in a Visual Studio project.
author: JonDouglas
ms.author: jodou
ms.date: 08/19/2022
ms.topic: quickstart
---

# Quickstart: Install and use a NuGet package in Visual Studio (Windows only)

> *NuGet* packages contain Microsoft-supported reusable code that other developers have made available to you for use in your projects. You can install NuGet packages in a Microsoft Visual Studio project by using the NuGet Package Manager, the Package Manager Console, or the .NET CLI. This article demonstrates how to create a Windows Presentation Foundation (WPF) project with the popular `Newtonsoft.Json` package. The same process applies to any other .NET or .NET Core project.

Reviewer: > *NuGet* packages contain Microsoft-supported reusable code that other developers have made available to you for use in your projects

I don't imagine we want to say they contain Microsoft-supported reusable code here do we?
It could be interpreted to imply that all packages are supported by MS which isn't true.

Reviewer: Is replacing `dotnet` with `.NET CLI` in this case the correct thing? `dotnet` refers to the exact CLI command, while `.NET` is just a generic name for everything .NET.

Reviewer 2: I agree "Microsoft-supported" should be removed. .NET CLI looks like the official name for the tool, so in this case, it looks good. I like having the link though.

Reviewer 3: What about `nuget.exe`?

Reviewer: nuget.exe's installation into project is not great, so probably a good idea not to recommend it tbh.

Author: We're discouraged from using links in quickstarts, especially in the intro paragraphs, so that the user stays focused. I could make an exception, though.
If we're not recommending nuget.exe/nuget.org, should I remove the TIP near the beginning? (To find a NuGet package, start with *nuget.org*... )

Reviewer: We wouldn't want to recommend nuget.exe in our quickstart here. VS and dotnet.exe support full package installation.

Nuget.org is the service that provides packages so that's ok.

After you install a package, you can then make a reference to it in your code with the `using <namespace>` statement, where \<namespace\> is the name of package you're using. After you've made a reference, you can then call the package through its API.

The article is for Windows users only. If you're using Visual Studio for Mac, see [Install and use a package in Visual Studio for Mac](install-and-use-a-package-in-visual-studio-mac.md).

> [!TIP]

> > To find a NuGet package, start with *nuget.org*. Browsing for nuget.org is how .NET developers typically find components they can reuse in their own applications. You can do a search of nuget.org directly or find and install packages within Visual Studio as shown in this article. For more information, see [Find and evaluate NuGet packages](../consume-packages/finding-and-choosing-packages.md).

Reviewer: > Browsing for nuget.org

Probably supposed to be `Browsing nuget.org`

## Prerequisites

- Install Visual Studio 2022 for Windows with the .NET desktop development workload.

  You can install the 2022 Community edition for free from [visualstudio.microsoft.com](https://visualstudio.microsoft.com/), or use the Professional or Enterprise edition.

> ## Create a WPF project

Reviewer: I think it'd be great for these quick starts to use the modern project templates, such as ones that are .NET SDK based, https://github.com/NuGet/docs.microsoft.com-nuget/issues/2853.

I'm not asking/suggesting a change here, mostly just linking the issues.

Reviewer 2: It looks like you can create a WPF template for .NET instead of .NET Framework. It would be an improvement to use the newer templates.

Reviewer 2: Please use **Create a project** for the heading instead of **Create a WPF project**. We want to avoid implying that a WPF project is a requirement rather than just an example.

Author: For the WPF template for .NET (which I think is WPF Application - A project for creating a .NET Application), should I tell the user to select the .NET Core 3.1 or .NET 6.0 framework during the install?

You can install a NuGet package in any .NET project, if that package supports the same target framework as the project.

> For this quickstart, create a WPF app project:

Reviewer 3: If we're no longer specifying WPF, maybe you could just say "(eg., WPF, Class Library, etc)" ?

Author: I still plan to specify WPF, just use a different template. It's a quickstart, so we're supposed to give the user a very specific set of instructions.

1. In Visual Studio, select **File** > **New** > **Project**.

1. Enter *.NET* in the search box, select **WPF App (.NET Framework)** for C#, and then select **Next**.

1. In the **Configure your new project** window, optionally update the **Project name** and the **Solution name**. Accept the default value for **Framework**, and then select **Create**.

   Visual Studio creates the project, and it appears in [Solution Explorer](/visualstudio/ide/use-solution-explorer).

## Add the Newtonsoft.Json NuGet package

To install a NuGet package, you can use either the NuGet Package Manager or the Package Manager Console. Depending on your project format, the installation of a NuGet package records the dependency in either your project file or a *packages.config* file. For more information, see [Package consumption workflow](../consume-packages/overview-and-workflow.md).

### NuGet Package Manager

To use the [NuGet Package Manager](../consume-packages/install-use-packages-visual-studio.md) to install the `Newtonsoft.Json` package in Visual Studio, follow these steps:

> 1. In **Solution Explorer**, right-click **References** and choose **Manage NuGet Packages**.

Reviewer: nit: The menu command is always available on the project node as well. Might be a more natural thing to click there.

    :::image type="content" source="media/qs-use-01-manage-nuget-packages.png" alt-text="Screenshot showing Solution Explorer with References selected and Manage NuGet Packages highlighted in shortcut menu.":::

1. Choose **nuget.org** as the **Package source**.

1. From the **Browse** tab, search for *Newtonsoft.Json*, select **Newtonsoft.Json** in the list, and then select **Install**.

    :::image type="content" source="media/qs-use-02-install-package.png" alt-text="Screenshot showing the NuGet Package Manager window with the Newtonsoft.Json package selected.":::

> 1. (Visual Studio 2017 only) If you're prompted to select a package management format, select **PackageReference in project file**:

Reviewer: This behavior is true starting with 2017.  We should probably update this to reflect that.

    ![Screenshot showing the Choose NuGet Package Manager Format window with PackageReference in project file selected.](media/qs-use-03-package-management-format.png)

1. If you're prompted to verify the installation, select **OK**.

### Package Manager Console

To use the [Package Manager Console](../consume-packages/install-use-packages-powershell.md) to install the `Newtonsoft.Json` package in Visual Studio, follow these steps:

1. From Visual Studio, select **Tools** > **NuGet Package Manager** > **Package Manager Console**.

1. After the **Package Manager Console** pane opens, verify that the **Default project** drop-down list shows the project in which you want to install the package. If you have a single project in the solution, it's already selected.

    :::image type="content" source="media/qs-use-04-package-manager-console.png" alt-text="media/qs-use-04-package-manager-console.png":::

1. At the console prompt, enter the command `Install-Package Newtonsoft.Json`. For more information about this command, see [Install-Package](../reference/ps-reference/ps-ref-install-package.md).

>    The console window the shows output for the command. Errors typically indicate that the package isn't compatible with the project's target framework.

Reviewer 3: ```suggestion
   The console window shows output for the command. Errors typically indicate that the package isn't compatible with the project's target framework.
```

    ![Screenshot showing the initial output of the WPF app.](media/qs-use-05-wpf-app-start.png)

1. Select the button to see the contents of the `TextBlock` object replaced with JSON text.

    ![Screenshot showing the output of the WPF app after selecting the button.](media/qs-use-06-wpf-app-end.png)

## Related video

- [Install and Use a NuGet Package with Visual Studio](/shows/nuget-101/install-and-use-a-nuget-package-with-visual-studio-2-of-5/player)
- Find more NuGet videos on [Channel 9](/shows/nuget-101/) and [YouTube](https://www.youtube.com/playlist?list=PLdo4fOcmZ0oVLvfkFk8O9h6v2Dcdh2bh_).

## See also

For more information about NuGet, see the following articles:

- [What is NuGet?](../what-is-nuget.md)
- [Package consumption workflow](../consume-packages/overview-and-workflow.md)
- [Find and choose packages](../consume-packages/finding-and-choosing-packages.md)
- [Package references in project files](../consume-packages/package-references-in-project-files.md)
- [Install and use a package using the .NET CLI](install-and-use-a-package-using-the-dotnet-cli.md).
- [Newtonsoft.Json package](https://www.nuget.org/packages/newtonsoft.json)

## Next steps

> Congratulations on installing and using your first NuGet package. Advance to the next article to learn how to install and manage NuGet packages.

Reviewer 3: Didn't we already install a package?
```suggestion
Congratulations on installing and using your first NuGet package. Advance to the next article to learn more about installing and managing NuGet packages.
```
